File: server/src/main/java/io/github/hulang1024/chinesechessserver/domain/chinesechess/ChessAction.java
package io.github.hulang1024.chinesechessserver.domain.chinesechess;

import io.github.hulang1024.chinesechessserver.domain.chinesechess.rule.Chess;
import io.github.hulang1024.chinesechessserver.domain.chinesechess.rule.ChessEnum;
import io.github.hulang1024.chinesechessserver.domain.chinesechess.rule.ChessHost;
import io.github.hulang1024.chinesechessserver.domain.chinesechess.rule.ChessPos;
import lombok.Data;

@Data
public class ChessAction {
    private ChessHost chessHost;
    private ChessEnum chessType;
    private ChessPos fromPos;
    private ChessPos toPos;
    private Chess eatenChess;
}

File: client/src/rulesets/chinesechess/ChessAction.ts
import Chess from "./rule/Chess";
import ChessPos from "./rule/ChessPos";
import ChessHost from "../chess_host";

export default class ChessAction {
  chessHost: ChessHost;

  chessType: any;

  isFront: boolean;

  fromPos: ChessPos;

  toPos: ChessPos;

  eatenChess: Chess | null;
}
